robusta_krr/core/integrations/prometheus/metrics_service/prometheus_metrics_service.py

```
@@ -199,6 +215,82 @@ async def get_history_range(self, history_duration: timedelta) -> tuple[datetime
+        return self.get_prometheus_cluster_label().replace(",", "")
+
+    async def _metric_has_data(self, metric: str) -> bool:
+        """
+        Checks whether a metric currently exists in this Prometheus.
+        Used to auto-detect which metric naming dialect this cluster exposes.
+        """
+
+        query = f"count({metric}{{ {self._single_cluster_label()} }})"
+        try:
+            return bool(await self.query(query))
+        except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)